# LAB 01: LAYOFFS 2023

## Mô tả đồ án:
- Từ nguồn dữ liệu lấy được từ Kaggle về vấn đề sa thải nhân viên ở các công ty trên toàn thế giới, đồ án này nhằm mục đích phân tích dữ liệu đã có để có được cái nhìn khách quan hơn về vấn đề này.

---

### Import các thư viện

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

ModuleNotFoundError: No module named 'seaborn'

---

### <span style="color:yellow"> A. Thu thập dữ liệu</span>


#### Ngữ cảnh thực hiện tìm kiếm dữ liệu

#### Chủ đề

#### Có cần phải tiền xử lý dữ liệu hay không và nếu có thì nhóm sinh viên cần phải xử lý như thế nào?

#### Nguồn
Dữ liệu được lấy trên Kaggle dưới file csv, đường link dẫn đến  [Ở đây](https://www.kaggle.com/datasets/swaptr/layoffs-2022?resource=download)

#### Người ta có cho phép sử dụng dữ liệu như thế này hay không?

#### Người ta đã thu thập dữ liệu này như thế nào? Phương pháp thực hiện là gì?

#### Đọc dữ liệu

In [5]:
df=pd.read_csv('./datasets/layoffs.csv')
df.head()

company       location     industry  total_laid_off  percentage_laid_off  \
0  Atlassian         Sydney        Other           500.0                 0.05   
1   SiriusXM  New York City        Media           475.0                 0.08   
2     Alerzo         Ibadan       Retail           400.0                  NaN   
3     UpGrad         Mumbai    Education           120.0                  NaN   
4       Loft      Sao Paulo  Real Estate           340.0                 0.15   

         date     stage        country  funds_raised  
0  2023-03-06  Post-IPO      Australia         210.0  
1  2023-03-06  Post-IPO  United States         525.0  
2  2023-03-06  Series B        Nigeria          16.0  
3  2023-03-06   Unknown          India         631.0  
4  2023-03-03   Unknown         Brazil         788.0

---

### <span style="color:yellow"> B. Khám phá dữ liệu</span>


#### Mỗi dòng có ý nghĩa gì? Có vấn đề các dòng có ý nghĩa khác nhau không?

In [7]:
df['stage']

<bound method Series.keys of 0       Post-IPO
1       Post-IPO
2       Series B
3        Unknown
4        Unknown
          ...   
2352     Unknown
2353        Seed
2354    Series C
2355    Series A
2356    Post-IPO
Name: stage, Length: 2357, dtype: object>

Mỗi dòng chứa dữ liệu về một công ty và các thông tin liên quan đến công ty đó như:
    Company: Tên của công ty
    Location: Thành phố nơi công ty đặt trụ sở
    Industry: Loại ngành mà công ty hoạt động
    Total_laid_off: Tổng số nhân viên bị sa thải bởi công ty
    Percentage_laid_off: Tỷ lệ phần trăm nhân viên bị sa thải bởi công ty
    Date: Ngày công bố thông tin về việc sa thải
    Stage: Giai đoạn của công ty (Post-IPO, Series B, Unknown)
    Country: Quốc gia nơi công ty đặt trụ sở
    Funds_raised: Tổng số tiền gây quỹ bởi công ty (tính bằng triệu đô la)

ví dụ: dòng 1: doanh nghiệp SiriusXM có trụ sở tại New York City và hoạt động trong ngành truyền thông, đã sa thải 475 nhân viên, chiếm 8% tổng số nhân viên của họ tính tới ngày 6 tháng 3 năm 2023. Công ty cũng đang ở giai đoạn Post-IPO và có trụ sở tại Hoa Kỳ. Họ đã quyên góp được 525 triệu đô la.

#### Mỗi cột có ý nghĩa gì?

#### Mỗi cột hiện đang có kiểu dữ liệu gì?

#### Có cột nào có kiểu dữ liệu chưa phù hợp để có thể xử lý tiếp hay không?

#### Có cần phải tiền xử lý dữ liệu hay không và nếu có thì nhóm sinh viên cần phải xử lý như thế nào?

In [ ]:
# nếu cần tiền xử lí